packages/editor/src/browser/semantic-highlight/semantic-highlighting-service.ts

```diff
@@ -0,0 +1,39 @@
+/********************************************************************************
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { Position, Range } from 'vscode-languageserver-types';
+import URI from '@theia/core/lib/common/uri';
+import { Disposable } from '@theia/core/lib/common/disposable';
+
+@injectable()
+export class SemanticHighlightingService implements Disposable {
```

Comment:
We should add some comment, explaining why this impl is entirely empty. It's ok to have a pointer to the monaco implementation in it.

Reply:
I can move the entire implementation from `@theia/monaco` to `@theia/editor` if I can expose the `id` property for the editor decoration.

The API for the `EditorDecoration` lacks this information.